flink-runtime/src/main/java/org/apache/flink/runtime/minicluster/MiniCluster.java

```diff
@@ -178,6 +181,13 @@ public URI getRestAddress() {
 		}
 	}
 
+	public HighAvailabilityServices getHighAvailabilityServices() {
+		synchronized (lock) {
+			checkState(running, "MiniCluster is not yet running.");
```

Comment:
I am curious about the lock in this place. Is not such a scene as follow ?

Some tests or ITCase running in maven test simultaneously. And might be change this object at the same time. So, that is why we should have a lock here ?

Reply:
Because all state accesses to the MiniCluster are guarded since it can be used by potentially multiple threads.